automation/features/step-definitions/staking.steps.js

```diff
@@ -48,23 +68,323 @@ When(/^I click on use maximum button$/, () => {
     reverse: false,
     timeoutMsg: "callout was not found",
   });
-  if (browser.getByTestId("associate-more-tokens-btn").isDisplayed()) {
-    browser.getByTestId("associate-more-tokens-btn").click();
+  if (stakingPage.associateMoreTokensBtn.isDisplayed()) {
+    stakingPage.associateMoreTokensBtn.click();
   } else {
-    browser.getByTestId("associate-tokens-btn").click();
+    stakingPage.associateTokensBtn.click();
   }
```

Comment:
In staking page you have this:

```
    if (this.associateMoreTokensBtn.isDisplayed()) {
      this.associateMoreTokensBtn.click();
    } else {
      this.associateTokensBtn.click();
    }
```
Let's create a function in the staking page called "clickOnAssociation" or similar that does this and just call it here. Removes the duplicated logic :D

Reply:
raised as issue